### PR TITLE
Cleanup TS warnings and improve types | Closes #2280

### LINKED ISSUE #2280: Cleanup TS Warnings
The following errors are issued when performing a build when running the new webpack config

```bash
./assets/prototype/domain/shared/entities/datetimes/predicates/filters/soldOutOnly/index.ts
  Line 10:19:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/datetimes/predicates/filters/allDates/index.ts
  Line 12:25:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/eventEditor/context/EventEditorEventIdProvider.tsx
  Line 12:36:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/datetimes/predicates/filters/aboveCapacity/index.ts
  Line 13:19:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/datetimes/predicates/filters/expiredOnly/index.ts
  Line 13:19:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/datetimes/predicates/capacityAtOrAbove.ts
  Line 13:27:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/tickets/predicates/filters/nextOnSaleOrPendingOnly.ts
  Line 13:33:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/datetimes/predicates/filters/belowCapacity/validInfiniteCapacityLimit.ts
  Line 15:36:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/predicates/shared/selectionPredicates/index.ts
  Line 16:30:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 17:30:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/tickets/predicates/filters/percentSoldBelow/index.ts
  Line 18:26:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 19:19:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/eventEditor/tickets/ticketsList/filterBar/controls/TicketsChainedButton.js
  Line 21:30:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/tickets/predicates/filters/percentSoldAtOrAbove/index.ts
  Line 22:30:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 23:15:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 30:19:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/datetimes/predicates/sorters.ts
  Line 24:17:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/prices/predicates/sortingPredicates.js
  Line 24:39:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/tickets/predicates/filters/index.ts
  Line 31:17:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/datetimes/predicates/filters/index.ts
  Line 33:17:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/application/utilities/text/i18n.ts
  Line 49:8:    Missing return type on function      @typescript-eslint/explicit-function-return-type
  Line 79:1:    Missing return type on function      @typescript-eslint/explicit-function-return-type
  Line 97:34:   'domain' is defined but never used   @typescript-eslint/no-unused-vars
  Line 113:8:   Missing return type on function      @typescript-eslint/explicit-function-return-type
  Line 113:20:  'text' is defined but never used     @typescript-eslint/no-unused-vars
  Line 113:26:  'context' is defined but never used  @typescript-eslint/no-unused-vars
  Line 113:35:  'domain' is defined but never used   @typescript-eslint/no-unused-vars
  Line 131:8:   Missing return type on function      @typescript-eslint/explicit-function-return-type
  Line 150:8:   Missing return type on function      @typescript-eslint/explicit-function-return-type
  Line 165:8:   Missing return type on function      @typescript-eslint/explicit-function-return-type
  Line 165:25:  'format' is defined but never used   @typescript-eslint/no-unused-vars
  Line 165:36:  'args' is defined but never used     @typescript-eslint/no-unused-vars

./assets/prototype/application/hooks/useRect.js
  Line 4:17:   Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 29:11:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 40:10:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 48:1:   Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/tickets/predicates/updatePredicates.js
  Line 4:33:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/ui/RelationsSelector.tsx
  Line 5:44:    Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 41:14:   'e' is defined but never used    @typescript-eslint/no-unused-vars
  Line 60:29:   Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 69:18:   Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 76:16:   Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 85:23:   Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 96:29:   Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 105:25:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 116:31:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 125:22:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 135:25:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 149:25:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 160:25:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 173:24:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 193:28:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 211:26:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 223:23:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 238:14:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 266:17:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/tickets/predicates/filters/onSaleOnly.ts
  Line 7:20:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/tickets/predicates/filters/soldOutOnly.ts
  Line 7:21:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/tickets/predicates/filters/pendingOnly.ts
  Line 7:21:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/tickets/predicates/filters/expiredOnly.ts
  Line 7:21:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/tickets/predicates/filters/archivedOnly/index.ts
  Line 7:22:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/datetimes/predicates/filters/recentlyExpiredOnly/index.ts
  Line 8:19:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/tickets/predicates/filters/onSaleAndPending.ts
  Line 8:26:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/tickets/predicates/selectionPredicates.js
  Line 8:30:   Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 10:34:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 11:39:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 13:43:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 15:45:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/prices/predicates/selectionPredicates.js
  Line 9:29:   Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 24:29:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 25:31:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 26:31:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 29:34:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 30:25:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 33:33:  Missing return type on function  @typescript-eslint/explicit-function-return-type
  Line 36:45:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/domain/shared/entities/tickets/predicates/isValidEntityOrArchive.ts
  Line 9:32:  Missing return type on function  @typescript-eslint/explicit-function-return-type

./assets/prototype/application/hooks/useIfMounted.ts
  Line 9:9:  Missing return type on function  @typescript-eslint/explicit-function-return-type
```

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR:
- Fixes some TS warnings
- Improves types
- Removes the unnecessary types where it's inferred e.g. `JSX.Element` when type is set as `React.FC`
- Closes #2280

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -36,7 +36,6 @@
 		"react/display-name": "off",
 		"@typescript-eslint/no-explicit-any": "off",
 		"@typescript-eslint/no-use-before-define": "off",
-		"@typescript-eslint/explicit-function-return-type": "off",
```
Comment (nicolad) [inline .eslintrc.json]: :+1: :taco: :medal_sports:

File: assets/src/application/services/apollo/mutations/useEntityMutation.ts [hunk 2]
```diff
@@ -49,8 +49,8 @@ const useEntityMutation = (type: EntityType, id?: string): EntityMutation => {
 	 */
 	const getMutationOptions = (mutationType: MutationType, input: MutationInput = {}): OperationVariables => {
 		// e.g. "datetimeMutator"
-		const key = `${type.toLowerCase()}Mutator`;
-		const mutator: Mutator = mutators[key];
+		const key = `${type.toLowerCase()}Mutator` as keyof Mutators;
```
Comment (nicolad) [inline assets/src/application/services/apollo/mutations/useEntityMutation.ts]: :+1:

File: assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/hooks/calculateBasePrice.ts [hunk 3]
```diff
@@ -21,10 +21,10 @@ const calculateBasePrice = (state: TpcFormData): TpcFormData => {
 	// we're calculating the base price so we don't want to include it in the calculations
 	const withoutBasePrice = filter<TpcPriceModifier>(isNotBasePrice, allPrices);
 	// and the last element should be the "NEW_PRICE" row and we don't want it either
-	const justModifiers: TpcPriceModifier[] = filter<TpcPriceModifier>(notNewPrice, withoutBasePrice);
-	const sortedModifiers: TpcPriceModifier[] = sortByPriceOrderIdDesc(justModifiers);
+	const justModifiers = filter<TpcPriceModifier>(notNewPrice, withoutBasePrice);
+	const sortedModifiers = sortByPriceOrderIdDesc(justModifiers);
 	// now extract the value for "total" or set to 0
-	const ticketTotal: number = propOr(0, 'price', ticket);
+	const ticketTotal = propOr<number, Ticket, number>(0, 'price', ticket);
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/hooks/calculateBasePrice.ts]: not sure I understood what last `number` do

Comment (manzoorwanijk) [reply assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/hooks/calculateBasePrice.ts]: Second `number` sets the return type:
```ts
propOr<T, U, V>(val: T, p: string, obj: U): V;
```

File: assets/src/application/services/apollo/mutations/useEntityMutation.ts [hunk 4]
```diff
@@ -81,8 +81,8 @@ const useEntityMutation = (type: EntityType, id?: string): EntityMutation => {
 			// e.g. "createDatetime", "updateTicket"
 			const mutationName = `${mutationType.toLowerCase()}Espresso${type}`;
 			// Example result: { data: { deletePrice: { price : {...} } } }
-			const path: string[] = ['data', mutationName, `espresso${type}`];
-			const entity: any = pathOr({}, path, result);
+			const path = ['data', mutationName, `espresso${type}`];
+			const entity = pathOr<any>({}, path, result);
```
Comment (tn3rb) [inline assets/src/application/services/apollo/mutations/useEntityMutation.ts]: can also do something like 

```ts
pathOr<T extends Entity>({}, path, result);
```

if there is a common sub-type expected

File: assets/src/application/services/apollo/mutations/useEntityMutation.ts (hunk 2 above)
Comment (tn3rb) [reply assets/src/application/services/apollo/mutations/useEntityMutation.ts]: cool... so is this basically saying that any value matching a property name in the Mutators object is acceptable?  Kinda like an enum but not hardcoded values.

File: assets/src/application/services/apollo/mutations/useEntityMutation.ts (hunk 4 above)
Comment (manzoorwanijk) [reply assets/src/application/services/apollo/mutations/useEntityMutation.ts]: Better

**Pull request merged**